### ISSUE TITLE
Feature discussion: Adding a negation reactive expression

### ISSUE THREAD

**Issue by AlexGustafsson** [fusetools/fuselibs-public#915]
If I'm correct, there's currently no direct way to negate a boolean value in UX Expressions. I'm suggesting an expression, `not ( bool ) : bool` that negates a given boolean value.

One of the simpler use cases I've found is probably best described through code;

```XML
<Rectangle ux:Class="Toggle" Width="50" Height="50" Background="Blue" State="False">
  <bool ux:Property="State" />
  <Tapped>
    <Set Target="this.State" Value="{not({Property State})}" />
   </Tapped>
</Panel>
```

_Note: This might not be the best use case scenario since there is an actual Toggle class, but the point still stands._

I have working code that I could push, given that the feature is wanted.

_Note: I've chosen to implement the functionality as a function rather than an arithmetic expression (such as `!`) as I find it more fit for the readable nature of UX markup._

**Comment by kusma** [comment]
Doen't `== false` or `!= true` work already?

**Comment by AlexGustafsson** [comment]
It does not seem to work.

```XML
<Rectangle ux:Class="Toggle" Width="50" Height="50" Background="Blue" State="False">
  <bool ux:Property="State" />
  <Tapped>
    <Set Target="this.State" Value="{{Property State} == true}" />
   </Tapped>
</Panel>
```

```XML
<Rectangle ux:Class="Toggle" Width="50" Height="50" Background="Blue" State="False">
  <bool ux:Property="State" />
  <Tapped>
    <Set Target="this.State" Value="{true == false}" />
   </Tapped>
</Panel>
```

The above two examples both throw 'Object reference not set to an instance of an object' without further detail.

I can not seem to find logical expressions in the documentation either.

Am I missing something?

**Comment by kusma** [comment]
@AlexGustafsson: Do you have a stacktrace for that?

**Comment by AlexGustafsson** [comment]
This is for `<Set Target="this.State" Value="{true == false}" />`

```
Fuse 1.6.0 (build 15070)
Starting preview for //Users/alexgustafsson/Documents/Apps/Fuse/HomeComponents/HomeComponents.unoproj, target Local
Opening '//Users/alexgustafsson/Documents/Apps/Fuse/HomeComponents/HomeComponents.unoproj'
Configuring

2,60 s

Compiling syntax tree

1,69 s

Generating code and data

5,51 s


Build completed in 9,82 seconds
Fuse.ComputeException: Cannot 'EqualTo'' objects of type 'System.Boolean'' and 'System.Boolean'
  at Fuse.Marshal.EqualTo (System.Object a, System.Object b) <0x11011c400 + 0x00133> in <filename unknown>:0 
  at Fuse.Reactive.Equal.Compute (System.Object left, System.Object right, System.Object& result) <0x11011c370 + 0x00029> in <filename unknown>:0 
  at Fuse.Reactive.BinaryOperator.OnNewOperands (IListener listener, System.Object left, System.Object right) <0x11011c200 + 0x0004d> in <filename unknown>:0 
  at Fuse.Reactive.BinaryOperator+Subscription.UpdateOperands () <0x11011bdf0 + 0x000ca> in <filename unknown>:0 
  at Fuse.Reactive.BinaryOperator+Subscription.OnNewData (IExpression source, System.Object value) <0x11011bd00 + 0x000bf> in <filename unknown>:0 
  at Fuse.Reactive.InnerListener.Fuse.Reactive.IListener.OnNewData (IExpression source, System.Object value) <0x11011ba20 + 0x0026a> in <filename unknown>:0 
  at Fuse.Reactive.ConstantExpression.Subscribe (IContext context, IListener listener) <0x11011b8c0 + 0x000ec> in <filename unknown>:0 
  at Fuse.Reactive.BinaryOperator+Subscription.Init (IContext context) <0x11011b540 + 0x00257> in <filename unknown>:0 
  at Fuse.Reactive.BinaryOperator+Subscription.Create (Fuse.Reactive.BinaryOperator bo, IContext context, IListener listener) <0x11011b400 + 0x0006d> in <filename unknown>:0 
  at Fuse.Reactive.BinaryOperator.Subscribe (IContext context, IListener listener) <0x11011b2d0 + 0x00031> in <filename unknown>:0 
  at Fuse.Reactive.ExpressionBinding.OnRooted () <0x10fddeb80 + 0x001af> in <filename unknown>:0 
  at Fuse.Reactive.DataBinding.OnRooted () <0x10fdde9f0 + 0x0002b> in <filename unknown>:0 
  at Fuse.Binding.Root (Fuse.Node parent) <0x10fdde8e0 + 0x000d5> in <filename unknown>:0 
  at Fuse.Node.RootBindings () <0x10fdb87b0 + 0x000bc> in <filename unknown>:0 
  at Fuse.Node.OnRooted () <0x10fdb8750 + 0x0001b> in <filename unknown>:0 
  at Fuse.NodeGroupBase.OnRooted () <0x110109ac0 + 0x00017> in <filename unknown>:0 
  at Fuse.Triggers.Trigger.OnRooted () <0x110109950 + 0x00017> in <filename unknown>:0 
  at Fuse.Gestures.ClickerTrigger.OnRooted () <0x11011b230 + 0x00017> in <filename unknown>:0 
  at Fuse.Gestures.Tapped.OnRooted () <0x11011b160 + 0x00017> in <filename unknown>:0 
  at Fuse.Node.RootInternalImpl (Fuse.Visual parent) <0x10fdb7df0 + 0x0021a> in <filename unknown>:0 
  at Fuse.Node.RootInternal (Fuse.Visual parent) <0x10fdb7ce0 + 0x0003f> in <filename unknown>:0 
  at Fuse.Visual.OnRooted () <0x10fdb82a0 + 0x001df> in <filename unknown>:0 
  at Fuse.Elements.Element.OnRooted () <0x10fdbb960 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Control.OnRooted () <0x10fdbb8d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.LayoutControl.OnRooted () <0x10fdbb850 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Shape.OnRooted () <0x110106030 + 0x0002b> in <filename unknown>:0 
  at Fuse.Node.RootInternalImpl (Fuse.Visual parent) <0x10fdb7df0 + 0x0021a> in <filename unknown>:0 
  at Fuse.Node.RootInternal (Fuse.Visual parent) <0x10fdb7ce0 + 0x0003f> in <filename unknown>:0 
  at Fuse.Visual.OnRooted () <0x10fdb82a0 + 0x001df> in <filename unknown>:0 
  at Fuse.Elements.Element.OnRooted () <0x10fdbb960 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Control.OnRooted () <0x10fdbb8d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.LayoutControl.OnRooted () <0x10fdbb850 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Panel.OnRooted () <0x10fdbb7d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Node.RootInternalImpl (Fuse.Visual parent) <0x10fdb7df0 + 0x0021a> in <filename unknown>:0 
  at Fuse.Node.RootInternal (Fuse.Visual parent) <0x10fdb7ce0 + 0x0003f> in <filename unknown>:0 
  at Fuse.Visual.OnRooted () <0x10fdb82a0 + 0x001df> in <filename unknown>:0 
  at Fuse.Elements.Element.OnRooted () <0x10fdbb960 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Control.OnRooted () <0x10fdbb8d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.LayoutControl.OnRooted () <0x10fdbb850 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Panel.OnRooted () <0x10fdbb7d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Node.RootInternalImpl (Fuse.Visual parent) <0x10fdb7df0 + 0x0021a> in <filename unknown>:0 
  at Fuse.Node.RootInternal (Fuse.Visual parent) <0x10fdb7ce0 + 0x0003f> in <filename unknown>:0 
  at Fuse.Visual.OnRooted () <0x10fdb82a0 + 0x001df> in <filename unknown>:0 
  at Fuse.Elements.Element.OnRooted () <0x10fdbb960 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Control.OnRooted () <0x10fdbb8d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.LayoutControl.OnRooted () <0x10fdbb850 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Panel.OnRooted () <0x10fdbb7d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Node.RootInternalImpl (Fuse.Visual parent) <0x10fdb7df0 + 0x0021a> in <filename unknown>:0 
  at Fuse.Node.RootInternal (Fuse.Visual parent) <0x10fdb7ce0 + 0x0003f> in <filename unknown>:0 
  at Fuse.Node.Relate (Fuse.Visual parent, Fuse.Node child) <0x10fdbb6c0 + 0x0004c> in <filename unknown>:0 
  at Fuse.Visual.OnAdded (Fuse.Node b) <0x10fdbb390 + 0x000e1> in <filename unknown>:0 
  at Fuse.Visual.Add (Fuse.Node item) <0x10fdbad00 + 0x00051> in <filename unknown>:0 
  at Fuse.Controls.Container.OnNodeAdded (Fuse.Node n) <0x110118f00 + 0x000e7> in <filename unknown>:0 
  at Uno.Collections.RootableList`1[T].OnAdded (Uno.Collections.T item) <0x1100da780 + 0x0006d> in <filename unknown>:0 
  at Uno.Collections.RootableList`1[T].RootSubscribe (System.Action`1 added, System.Action`1 removed) <0x1101094e0 + 0x000cf> in <filename unknown>:0 
  at Fuse.Controls.Container.OnRooted () <0x110118da0 + 0x0010f> in <filename unknown>:0 
  at Fuse.Node.RootInternalImpl (Fuse.Visual parent) <0x10fdb7df0 + 0x0021a> in <filename unknown>:0 
  at Fuse.Node.RootInternal (Fuse.Visual parent) <0x10fdb7ce0 + 0x0003f> in <filename unknown>:0 
  at Fuse.Visual.OnRooted () <0x10fdb82a0 + 0x001df> in <filename unknown>:0 
  at Fuse.Elements.Element.OnRooted () <0x10fdbb960 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Control.OnRooted () <0x10fdbb8d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.LayoutControl.OnRooted () <0x10fdbb850 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Panel.OnRooted () <0x10fdbb7d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Node.RootInternalImpl (Fuse.Visual parent) <0x10fdb7df0 + 0x0021a> in <filename unknown>:0 
  at Fuse.Node.RootInternal (Fuse.Visual parent) <0x10fdb7ce0 + 0x0003f> in <filename unknown>:0 
  at Fuse.Visual.OnRooted () <0x10fdb82a0 + 0x001df> in <filename unknown>:0 
  at Fuse.Elements.Element.OnRooted () <0x10fdbb960 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Control.OnRooted () <0x10fdbb8d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.LayoutControl.OnRooted () <0x10fdbb850 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Panel.OnRooted () <0x10fdbb7d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Node.RootInternalImpl (Fuse.Visual parent) <0x10fdb7df0 + 0x0021a> in <filename unknown>:0 
  at Fuse.Node.RootInternal (Fuse.Visual parent) <0x10fdb7ce0 + 0x0003f> in <filename unknown>:0 
  at Fuse.Visual.OnRooted () <0x10fdb82a0 + 0x001df> in <filename unknown>:0 
  at Fuse.Elements.Element.OnRooted () <0x10fdbb960 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Control.OnRooted () <0x10fdbb8d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.LayoutControl.OnRooted () <0x10fdbb850 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Panel.OnRooted () <0x10fdbb7d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Node.RootInternalImpl (Fuse.Visual parent) <0x10fdb7df0 + 0x0021a> in <filename unknown>:0 
  at Fuse.Node.RootInternal (Fuse.Visual parent) <0x10fdb7ce0 + 0x0003f> in <filename unknown>:0 
  at Fuse.Node.Relate (Fuse.Visual parent, Fuse.Node child) <0x10fdbb6c0 + 0x0004c> in <filename unknown>:0 
  at Fuse.Visual.OnAdded (Fuse.Node b) <0x10fdbb390 + 0x000e1> in <filename unknown>:0 
  at Fuse.Visual.Add (Fuse.Node item) <0x10fdbad00 + 0x00051> in <filename unknown>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x106ab1530 + 0x000b7> in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <0x11011ced0 + 0x00038> in <filename unknown>:0 
  at Outracks.UnoHost.TryInvokeExtension.Try (System.Func`1 func) <0x10f967a50 + 0x0007d> in <filename unknown>:0 
  at Outracks.UnoHost.TryInvokeExtension.TryInvoke (System.Reflection.MethodInfo mi, System.Object obj, System.Object[] value) <0x10f9848a0 + 0x0019d> in <filename unknown>:0 
  at Outracks.UnoHost.Reflection.CallDynamic (System.Object obj, System.String methodName, System.Object[] arguments) <0x1100d7e30 + 0x0004a> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.UnoHostReflection.CallDynamic (System.Object obj, Outracks.Simulator.Bytecode.TypeMemberName methodName, System.Object[] arguments) <0x1100d7de0 + 0x00030> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.ScopeClosure.Evaluate (Outracks.Simulator.Bytecode.CallDynamicMethod m) <0x1100d7d50 + 0x0007e> in <filename unknown>:0 
  at Outracks.Simulator.Bytecode.Expression.Match[T] (System.Func`2 a1, System.Func`2 a2, System.Func`2 a3, System.Func`2 a4, System.Func`2 a5, System.Func`2 a6, System.Func`2 a7, System.Func`2 a8, System.Func`2 a9, System.Func`2 a10, System.Func`2 a11, System.Func`2 a12, System.Func`2 a13, System.Func`2 a15, System.Func`2 a16) <0x1100b89d0 + 0x004d6> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.ScopeClosure.Evaluate (Outracks.Simulator.Bytecode.Expression expression) <0x1100b8060 + 0x008d6> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.ScopeClosure.Execute (IEnumerable`1 statements) <0x1100b7cd0 + 0x000ea> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.ScopeClosure.Execute (Outracks.Simulator.Bytecode.Lambda lambda, System.Object[] arguments) <0x1100b7710 + 0x0030e> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.LambdaClosure.Func (System.Object a1) <0x1100c6900 + 0x00072> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.ScopeClosure.Execute (System.Object lambda, System.Object[] arguments) <0x1100c3200 + 0x00144> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.ScopeClosure.Evaluate (Outracks.Simulator.Bytecode.CallLambda i) <0x1100be9e0 + 0x00063> in <filename unknown>:0 
  at Outracks.Simulator.Bytecode.Expression.Match[T] (System.Func`2 a1, System.Func`2 a2, System.Func`2 a3, System.Func`2 a4, System.Func`2 a5, System.Func`2 a6, System.Func`2 a7, System.Func`2 a8, System.Func`2 a9, System.Func`2 a10, System.Func`2 a11, System.Func`2 a12, System.Func`2 a13, System.Func`2 a15, System.Func`2 a16) <0x1100b89d0 + 0x003e8> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.ScopeClosure.Evaluate (Outracks.Simulator.Bytecode.Expression expression) <0x1100b8060 + 0x008d6> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.ScopeClosure.Execute (IEnumerable`1 statements) <0x1100b7cd0 + 0x000ea> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.ScopeClosure.Execute (Outracks.Simulator.Bytecode.Lambda lambda, System.Object[] arguments) <0x1100b7710 + 0x0030e> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.LambdaClosure.Func (System.Object a1) <0x1100c6900 + 0x00072> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.LambdaClosure.Action (System.Object a1) <0x1100c68c0 + 0x00020> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.ScopeClosure.Execute (System.Object lambda, System.Object[] arguments) <0x1100c3200 + 0x0168f> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.ScopeClosure.Evaluate (Outracks.Simulator.Bytecode.CallLambda i) <0x1100be9e0 + 0x00063> in <filename unknown>:0 
  at Outracks.Simulator.Bytecode.Expression.Match[T] (System.Func`2 a1, System.Func`2 a2, System.Func`2 a3, System.Func`2 a4, System.Func`2 a5, System.Func`2 a6, System.Func`2 a7, System.Func`2 a8, System.Func`2 a9, System.Func`2 a10, System.Func`2 a11, System.Func`2 a12, System.Func`2 a13, System.Func`2 a15, System.Func`2 a16) <0x1100b89d0 + 0x003e8> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.ScopeClosure.Evaluate (Outracks.Simulator.Bytecode.Expression expression) <0x1100b8060 + 0x008d6> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.ScopeClosure.Execute (IEnumerable`1 statements) <0x1100b7cd0 + 0x000ea> in <filename unknown>:0 
  at Outracks.Simulator.Runtime.ScopeClosure.Execute (Outracks.Simulator.Bytecode.Lambda lambda, System.Object[] arguments) <0x1100b7710 + 0x0030e> in <filename unknown>:0 
  at Outracks.Simulator.Client.VirtualMachine.Execute (IReflection reflection, Outracks.Simulator.Bytecode.Lambda lambda, System.Object[] arguments) <0x1100b7330 + 0x000c6> in <filename unknown>:0 
  at Outracks.Simulator.Client.VirtualMachine.Execute (Outracks.Simulator.Bytecode.Lambda lambda, System.Object[] arguments) <0x1100b72e0 + 0x0002f> in <filename unknown>:0 
  at Outracks.Simulator.Client.Reifying.OnEnterState () <0x1100bda90 + 0x001f1> in <filename unknown>:0 
  at Outracks.Simulator.Client.Application.SetState (Outracks.Simulator.Client.State nextState) <0x10fdd7420 + 0x00078> in <filename unknown>:0 
  at Outracks.Simulator.Client.Application.OnUpdate () <0x10fdd6f10 + 0x0014a> in <filename unknown>:0 
  at Fuse.App.Update () <0x10fdd6d10 + 0x0003d> in <filename unknown>:0
System.Exception: Node already has a parent - can only have one
  at Fuse.Visual.Children_MakeParent (Fuse.Visual parent, Fuse.Node child) <0x10fdbb310 + 0x00072> in <filename unknown>:0 
  at Fuse.Visual.Children_Add (Fuse.Node n) <0x10fdbafe0 + 0x00042> in <filename unknown>:0 
  at Fuse.Visual.Add (Fuse.Node item) <0x10fdbad00 + 0x0003a> in <filename unknown>:0 
  at Outracks.Simulator.Client.Application.ResetEverything (Boolean initial) <0x10fdc0000 + 0x007b7> in <filename unknown>:0 
  at Outracks.Simulator.Client.Application.OnUpdate () <0x10fdd6f10 + 0x000c1> in <filename unknown>:0 
  at Fuse.App.Update () <0x10fdd6d10 + 0x0003d> in <filename unknown>:0
Fuse.ComputeException: Cannot 'EqualTo'' objects of type 'System.Boolean'' and 'System.Boolean'
  at Fuse.Marshal.EqualTo (System.Object a, System.Object b) <0x11011c400 + 0x00133> in <filename unknown>:0 
  at Fuse.Reactive.Equal.Compute (System.Object left, System.Object right, System.Object& result) <0x11011c370 + 0x00029> in <filename unknown>:0 
  at Fuse.Reactive.BinaryOperator.OnNewOperands (IListener listener, System.Object left, System.Object right) <0x11011c200 + 0x0004d> in <filename unknown>:0 
  at Fuse.Reactive.BinaryOperator+Subscription.UpdateOperands () <0x11011bdf0 + 0x000ca> in <filename unknown>:0 
  at Fuse.Reactive.BinaryOperator+Subscription.OnNewData (IExpression source, System.Object value) <0x11011bd00 + 0x000bf> in <filename unknown>:0 
  at Fuse.Reactive.InnerListener.Fuse.Reactive.IListener.OnNewData (IExpression source, System.Object value) <0x11011ba20 + 0x0026a> in <filename unknown>:0 
  at Fuse.Reactive.ConstantExpression.Subscribe (IContext context, IListener listener) <0x11011b8c0 + 0x000ec> in <filename unknown>:0 
  at Fuse.Reactive.BinaryOperator+Subscription.Init (IContext context) <0x11011b540 + 0x00257> in <filename unknown>:0 
  at Fuse.Reactive.BinaryOperator+Subscription.Create (Fuse.Reactive.BinaryOperator bo, IContext context, IListener listener) <0x11011b400 + 0x0006d> in <filename unknown>:0 
  at Fuse.Reactive.BinaryOperator.Subscribe (IContext context, IListener listener) <0x11011b2d0 + 0x00031> in <filename unknown>:0 
  at Fuse.Reactive.ExpressionBinding.OnRooted () <0x10fddeb80 + 0x001af> in <filename unknown>:0 
  at Fuse.Reactive.DataBinding.OnRooted () <0x10fdde9f0 + 0x0002b> in <filename unknown>:0 
  at Fuse.Binding.Root (Fuse.Node parent) <0x10fdde8e0 + 0x000d5> in <filename unknown>:0 
  at Fuse.Node.RootBindings () <0x10fdb87b0 + 0x000bc> in <filename unknown>:0 
  at Fuse.Node.OnRooted () <0x10fdb8750 + 0x0001b> in <filename unknown>:0 
  at Fuse.NodeGroupBase.OnRooted () <0x110109ac0 + 0x00017> in <filename unknown>:0 
  at Fuse.Triggers.Trigger.OnRooted () <0x110109950 + 0x00017> in <filename unknown>:0 
  at Fuse.Gestures.ClickerTrigger.OnRooted () <0x11011b230 + 0x00017> in <filename unknown>:0 
  at Fuse.Gestures.Tapped.OnRooted () <0x11011b160 + 0x00017> in <filename unknown>:0 
  at Fuse.Node.RootInternalImpl (Fuse.Visual parent) <0x10fdb7df0 + 0x0021a> in <filename unknown>:0 
  at Fuse.Node.RootInternal (Fuse.Visual parent) <0x10fdb7ce0 + 0x0003f> in <filename unknown>:0 
  at Fuse.Visual.OnRooted () <0x10fdb82a0 + 0x001df> in <filename unknown>:0 
  at Fuse.Elements.Element.OnRooted () <0x10fdbb960 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Control.OnRooted () <0x10fdbb8d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.LayoutControl.OnRooted () <0x10fdbb850 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Shape.OnRooted () <0x110106030 + 0x0002b> in <filename unknown>:0 
  at Fuse.Node.RootInternalImpl (Fuse.Visual parent) <0x10fdb7df0 + 0x0021a> in <filename unknown>:0 
  at Fuse.Node.RootInternal (Fuse.Visual parent) <0x10fdb7ce0 + 0x0003f> in <filename unknown>:0 
  at Fuse.Visual.OnRooted () <0x10fdb82a0 + 0x001df> in <filename unknown>:0 
  at Fuse.Elements.Element.OnRooted () <0x10fdbb960 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Control.OnRooted () <0x10fdbb8d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.LayoutControl.OnRooted () <0x10fdbb850 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Panel.OnRooted () <0x10fdbb7d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Node.RootInternalImpl (Fuse.Visual parent) <0x10fdb7df0 + 0x0021a> in <filename unknown>:0 
  at Fuse.Node.RootInternal (Fuse.Visual parent) <0x10fdb7ce0 + 0x0003f> in <filename unknown>:0 
  at Fuse.Visual.OnRooted () <0x10fdb82a0 + 0x001df> in <filename unknown>:0 
  at Fuse.Elements.Element.OnRooted () <0x10fdbb960 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Control.OnRooted () <0x10fdbb8d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.LayoutControl.OnRooted () <0x10fdbb850 + 0x00017> in <filename unknown>:0 
  at Fuse.Controls.Panel.OnRooted () <0x10fdbb7d0 + 0x00017> in <filename unknown>:0 
  at Fuse.Node.RootInternalImpl (Fuse.Visual parent) <0x10fdb7df0 + 0x0021a> in <filename unknown>:0 
  at Fuse.Node.RootInternal (Fuse.Visual parent) <0x10fdb7ce0 + 0x0003f> in <filename unknown>:0 
  at Fuse.Node.Relate (Fuse.Visual parent, Fuse.Node child) <0x10fdbb6c0 + 0x0004c> in <filename unknown>:0 
  at Fuse.Visual.OnAdded (Fuse.Node b) <0x10fdbb390 + 0x000e1> in <filename unknown>:0 
  at Fuse.Visual.InsertNodesAfterImpl (Fuse.Node preceeder, IEnumerator`1 items, Boolean allowMove) <0x110121ca0 + 0x00254> in <filename unknown>:0 
  at Fuse.Visual.InsertNodesAfter (Fuse.Node preceeder, IEnumerator`1 items) <0x110121c60 + 0x00028> in <filename unknown>:0 
  at Fuse.Reactive.Instantiator.CompleteWindowItem (Fuse.Reactive.WindowItem wi, Int32 windowIndex) <0x11013d500 + 0x003fa> in <filename unknown>:0 
  at Fuse.Reactive.Instantiator.CompleteWindowItems (Boolean one) <0x11013d1c0 + 0x000bc> in <filename unknown>:0 
  at Fuse.Reactive.Instantiator.CompleteWindowItemsAction () <0x11013d180 + 0x0001d> in <filename unknown>:0 
  at Fuse.UpdateAction.Invoke () <0x110082c00 + 0x0005e> in <filename unknown>:0 
  at Fuse.UpdateManager.ProcessDeferredActions (Fuse.Stage stage, System.Collections.Generic.List`1& _exceptions) <0x110082980 + 0x00056> in <filename unknown>:0
```

**Comment by kusma** [comment]
Thanks. OK, so by "throw" you mean "give a diagnostic error". That's not as bad as I though.

Yeah, seems this is simply not implemented. That's a bit disappointing, so I gave it a go: #916. Hopefully this works for you :)

**Comment by AlexGustafsson** [comment]
Sorry for any confusion on the wording there, my bad.

I'll give it a go soon. Thank you for a prompt fix!

**Comment by AlexGustafsson** [comment]
@kusma Do you think that the negate operator `!` is in place as well? Although `value != true` etc. could be used, using `!value` or the like is more in line with what developers might expect from boolean operators (I assume). Having a negate operator also helps simplify the syntax needed for toggling.

**Comment by yupferris** [comment]
We already parse expressions with `!` actually, so if it doesn't work it's something else down the line. I'll take a look :)

**Comment by kusma** [comment]
OK, this has been merged, and is expected to arrive in Fuse 1.7.

**Comment by AlexGustafsson** [comment]
Is it supposed to work in the latest version of master?

I'm using Fuse v1.6.0 (build 15197) and the latest version of fuselibs (at time of writing this). Whenever I use `Value="!{Property State}"` or the like (usage as seen in tests) I get the following errors once per usage:

```
E8001: Type not found: Fuse.Reactive.LogicalNot
Error E8001: Type not found: Fuse.Reactive.LogicalNot
```

**Comment by kusma** [comment]
@AlexGustafsson: No, as I wrote above, this is expected to arrive in Fuse 1.7. Fuse 1.6 had already been shipped off to internal QA before this feature was added.

**Comment by AlexGustafsson** [comment]
@kusma My bad, I interpreted it as it was fixed in fuselibs but that the version won't be included with the current Fuse build until 1.7. Thanks for clearing that up.
